japicmp/src/main/java/japicmp/compat/CompatibilityChanges.java

```diff
@@ -378,6 +378,26 @@ public Integer callback(JApiClass superclass, Map<String, JApiClass> classMap, J
 					addCompatibilityChange(method, JApiCompatibilityChange.METHOD_NOW_FINAL);
 				}
 			}
+			if (isNotPrivate(method) && method.getChangeStatus() == JApiChangeStatus.REMOVED) {
+				List<Integer> returnValues = new ArrayList<>();
+				forAllSuperclasses(jApiClass, classMap, returnValues, new OnSuperclassCallback<Integer>() {
+					@Override
+					public Integer callback(JApiClass superclass, Map<String, JApiClass> classMap, JApiChangeStatus changeStatusOfSuperclass) {
+						for (JApiMethod superMethod : superclass.getMethods()) {
+							if (areMatching(superMethod, method)) {
+								if (method.getFinalModifier().getOldModifier().get() == FinalModifier.NON_FINAL
```

Comment:
Generally speaking I dislike this approach, because eventually it will require duplicating every single method compatibility check for the case where the method was moved up.
Ideally it would be possible to pass 2 arbitrary methods (old/new) into a compatibility routine, but I can't see an easy way to do that with the current architecture of encoding both the old and new versions of a method into a single `JApiMethod`.